prime-router/metadata/hl7_mapping/common/message-header.yml

```diff
@@ -4,27 +4,33 @@ elements:
     value: [ '%resource.extension(%`rsext-encoding-characters`).value' ]
     hl7Spec: [ MSH-2 ]
 
-  - name: sending-application-namespace-id-no-endpoint
-    value: [ '%resource.source.name.split(":").first()' ]
+  - name: sending-application-namespace-id-universal-id-type-known
+    condition: '%resource.source.endpoint.ofType(Url).exists()'
+    value: [ '%resource.source.name' ]
     hl7Spec: [ MSH-3-1 ]
 
-  - name: sending-application-universal-id-with-endpoint
-    condition: '%resource.source.endpoint.exists()'
+  - name: sending-application-namespace-id-universal-id-type-unknown
+    condition: '%resource.source.endpoint.where(extension(%`rsext-data-absent-reason`).value = "unknown").exists()'
+    value: [ '%resource.source.name.split("-").first()' ]
```

Comment:
the mapping asks that `-` and `:` are used to combine the three HD fields in certain cases. To build the HL7 again we have to split on those characters. That means we should consider them special and not allow them in incoming HL7 `MSH.3` or `MSH.24`. Another note for validation